test/pretty/atd.expected.json

```diff
@@ -1,12 +1,35 @@
 [
-  { "r1": "testing" }, { "r2": [ "Some", 2 ] }, { "r2": "None" },
-  { "r3": [ "Some", 3 ] }, {}, { "r4": true }, { "r5": [ "Some", 5 ] }, {},
-  { "r6": 6 }, {}, { "r7": -1000 }, { "r8": [ 1, 2, 3 ] }, [ "foo", "bar" ],
-  [], null, [ 1, 2, 3 ], 99, { "foo": 7, "bar": 8, "baz": 43 },
-  { "foo2": 5, "bar2": 6, "baz2": 41, "42": 42 }, [ 100, "foo" ],
-  [ 100, 200, 42 ], [ 100, 200, -1 ],
+  { "r1": "testing" },
+  { "r2": [ "Some", 2 ] },
+  { "r2": "None" },
+  { "r3": [ "Some", 3 ] },
+  {},
+  { "r4": true },
+  { "r5": [ "Some", 5 ] },
+  {},
+  { "r6": 6 },
+  {},
+  { "r7": -1000 },
+  { "r8": [ 1, 2, 3 ] },
+  [ "foo", "bar" ],
+  [],
+  null,
+  [ 1, 2, 3 ],
+  99,
+  { "foo": 7, "bar": 8, "baz": 43 },
+  { "foo2": 5, "bar2": 6, "baz2": 41, "42": 42 },
+  [ 100, "foo" ],
+  [ 100, 200, 42 ],
+  [ 100, 200, -1 ],
   [
-    "V1", "v22", [ "V3", "testing" ], [ "V44", 255 ], [ "V5", "None" ],
-    [ "V5", [ "Some", true ] ] ], { "v2": "A" }, { "v2": [ "B", 100 ] },
-  [ "C1", [ "C2", true ], [ "C2", false ] ], [ 50, 30, -1, 400 ]
+    "V1",
+    "v22",
+    [ "V3", "testing" ],
+    [ "V44", 255 ],
+    [ "V5", "None" ],
+    [ "V5", [ "Some", true ] ] ],
```

Comment:
Shouldn't the last closing `]` here be on its own line?

Reply:
I agree, it's surprising.

Reply:
@c-cube I'm trying to figure out what `inside_box` is for. From looking that the easy-format library, it seems that what we want to print JSON arrays is:
* an outer hv box (horizontal or vertical layout) rather than hov (current code when `not inside_box`).
* an inner box for the list body, that gets indented (hv or hov depending on the chosen style).

I'll try a few things.

Reply:
Changing the array's outer hov box to hv seems to have worked.